osmoutils/store_helper.go

```diff
@@ -137,3 +137,16 @@ func MustGetDec(store store.KVStore, key []byte) sdk.Dec {
 	MustGet(store, key, result)
 	return result.Dec
 }
+
+// GetIfFound gets key from store
+// returns a boolean indicating whether value exists for the given key and error
+func GetIfFound(store store.KVStore, key []byte, result proto.Message) (found bool, err error) {
```

Comment:
Can be changed later / can discuss in an issue, should not block this PR. Perhaps this should be called `HasWithType` ?

e.g. if we were masters of the store interface
```
store.HasProtoMsg("myKey", types.Pool)
```

Has to indicate its a bool (rather than assuming you'd get the object back)

OR is proto.Message here actually mutated? If so lets update the comment, but this doesn't actually make sense to me as an API.

Why don't just make a method called `Get` thats the same as MustGet, but returns an error instead of panics?